File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch10/C/P10-qImprop.cpp
// Evaluates improper integrals
#include <stdio.h>
#include <math.h>
#include "integral.h"

double Func1(double x) { return x * exp(-x*x); }
double Func2(double x) { return (x ? sin(x)/x : 1e0); }
double Func3(double x) { return 1e0/sqrt(x); }

int main()
{
   double eps, xinf;
   
   xinf = 1e10; eps = 1e-8;
   printf("I1 = %f xinf = %f\n",qImprop1(Func1,0e0,xinf,eps),xinf);
   printf("I2 = %f xinf = %f\n",qImprop1(Func2,0e0,xinf,eps),xinf);
   printf("I3 = %f\n",qImprop2(Func3,0e0,1e0,eps));
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch10/Python/P10-qImprop.py
# Evaluates improper integrals
from math import *
from integral import *

def Func1(x): return x * exp(-x*x)
def Func2(x): return sin(x)/x if x else 1e0
def Func3(x): return 1/sqrt(x)

# main

xinf = 1e10
eps = 1e-6
(s, xinf) = qImprop1(Func1,0e0,xinf,eps)
print("I1 = ",s," xinf = ",xinf)
(s, xinf) = qImprop1(Func2,0e0,xinf,eps)
print("I2 = ",s," xinf = ",xinf)
print("I3 = ",qImprop2(Func3,0e0,1e0,eps))
